skimage/feature/texture.py

```diff
@@ -264,6 +268,22 @@ def graycoprops(P, prop='contrast'):
         results = np.sqrt(asm)
     elif prop == 'ASM':
         results = np.sum(P**2, axis=(0, 1))
+    elif prop == 'mean':
+        I = np.array(range(num_level)).reshape((num_level, 1, 1, 1))
+        results = np.sum(I * P, axis=(0, 1))
+    elif prop == 'variance':
+        I = np.array(range(num_level)).reshape((num_level, 1, 1, 1))
+        mean = np.sum(I * P, axis=(0, 1))
+        results = np.sum(P * ((I - mean) ** 2), axis=(0, 1))
+    elif prop == 'std':
+        I = np.array(range(num_level)).reshape((num_level, 1, 1, 1))
+        mean = np.sum(I * P, axis=(0, 1))
```

Comment:
What do you think about turning this repeated code into a simple closure function like 

```python
def glcm_mean():
    I = np.arange(num_level).reshape((num_level, 1, 1, 1))
    mean = np.sum(I * P, axis=(0, 1))
    return I, mean
```

to reuse?

Reply:
It's a good idea, it was bit repetitive.